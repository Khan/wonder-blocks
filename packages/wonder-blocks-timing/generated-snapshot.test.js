// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-timing
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");

describe("wonder-blocks-timing", () => {
    it("example 1", () => {
        const Button = require("@khanacademy/wonder-blocks-button").default;
        const {IDProvider, View} = require("@khanacademy/wonder-blocks-core");

        class Unmounter extends React.Component {
            constructor() {
                super();
                this.state = {
                    mountKids: true,
                };
            }

            maybeRenderKids() {
                if (this.state.mountKids) {
                    return (
                        <React.Fragment>
                            <Button onClick={() => this.onClick()}>
                                Unmount
                            </Button>
                            {this.props.children}
                        </React.Fragment>
                    );
                } else {
                    return "Children unmounted";
                }
            }

            onClick() {
                this.setState({mountKids: false});
            }

            render() {
                return <View>{this.maybeRenderKids()}</View>;
            }
        }

        class MyNaughtyComponent extends React.Component {
            componentDidMount() {
                const {targetId} = this.props;
                let counter = 0;
                const domElement = document.getElementById(targetId);
                setInterval(() => {
                    domElement.innerText =
                        "Naughty interval logged: " + counter++;
                }, 200);
            }

            render() {
                return <View>NaughtyComponent here</View>;
            }
        }

        const example = (
            <IDProvider>
                {(id) => (
                    <View>
                        <Unmounter>
                            <MyNaughtyComponent targetId={id} />
                        </Unmounter>
                        <View>
                            <View id={id} />
                        </View>
                    </View>
                )}
            </IDProvider>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const {
            withActionScheduler,
        } = require("@khanacademy/wonder-blocks-timing");
        const Button = require("@khanacademy/wonder-blocks-button").default;
        const {IDProvider, View} = require("@khanacademy/wonder-blocks-core");

        class Unmounter extends React.Component {
            constructor() {
                super();
                this.state = {
                    mountKids: true,
                };
            }

            maybeRenderKids() {
                if (this.state.mountKids) {
                    return (
                        <React.Fragment>
                            <Button onClick={() => this.onClick()}>
                                Unmount
                            </Button>
                            {this.props.children}
                        </React.Fragment>
                    );
                } else {
                    return "Children unmounted";
                }
            }

            onClick() {
                this.setState({mountKids: false});
            }

            render() {
                return <View>{this.maybeRenderKids()}</View>;
            }
        }

        class MyGoodComponent extends React.Component {
            componentDidMount() {
                const {targetId, schedule} = this.props;
                let counter = 0;
                const domElement = document.getElementById(targetId);
                schedule.interval(() => {
                    domElement.innerText =
                        "Naughty interval logged: " + counter++;
                }, 200);
            }

            render() {
                return <View>GoodComponent here</View>;
            }
        }

        const MyGoodComponentWithScheduler = withActionScheduler(
            MyGoodComponent,
        );

        const example = (
            <IDProvider>
                {(id) => (
                    <View>
                        <Unmounter>
                            <MyGoodComponentWithScheduler targetId={id} />
                        </Unmounter>
                        <View>
                            <View id={id} />
                        </View>
                    </View>
                )}
            </IDProvider>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
