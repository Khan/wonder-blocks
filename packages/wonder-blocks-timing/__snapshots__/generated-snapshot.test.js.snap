// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-timing example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <button
      className=""
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "::MozFocusInner": Object {
            "border": 0,
          },
          ":focus": Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          },
          "alignItems": "center",
          "background": "#1865f2",
          "border": "none",
          "borderRadius": 4,
          "boxSizing": "border-box",
          "color": "#ffffff",
          "cursor": "pointer",
          "display": "inline-flex",
          "height": 40,
          "justifyContent": "center",
          "margin": 0,
          "outline": "none",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "position": "relative",
          "textDecoration": "none",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className=""
        style={
          Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "alignItems": "center",
            "display": "flex",
            "fontFamily": "Lato, sans-serif",
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": "20px",
            "overflow": "hidden",
            "pointerEvents": "none",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
          }
        }
      >
        Unmount
      </span>
    </button>
    <div
      className=""
      style={
        Object {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      NaughtyComponent here
    </div>
  </div>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      id="uid--0-wb-id"
      style={
        Object {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    />
  </div>
</div>
`;

exports[`wonder-blocks-timing example 2 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <button
      className=""
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "::MozFocusInner": Object {
            "border": 0,
          },
          ":focus": Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          },
          "alignItems": "center",
          "background": "#1865f2",
          "border": "none",
          "borderRadius": 4,
          "boxSizing": "border-box",
          "color": "#ffffff",
          "cursor": "pointer",
          "display": "inline-flex",
          "height": 40,
          "justifyContent": "center",
          "margin": 0,
          "outline": "none",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "position": "relative",
          "textDecoration": "none",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className=""
        style={
          Object {
            "MozOsxFontSmoothing": "grayscale",
            "WebkitFontSmoothing": "antialiased",
            "alignItems": "center",
            "display": "flex",
            "fontFamily": "Lato, sans-serif",
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": "20px",
            "overflow": "hidden",
            "pointerEvents": "none",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
          }
        }
      >
        Unmount
      </span>
    </button>
    <div
      className=""
      style={
        Object {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      GoodComponent here
    </div>
  </div>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      id="uid--1-wb-id"
      style={
        Object {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    />
  </div>
</div>
`;
